### PR TITLE
Basic version of collections

### PR DESCRIPTION
**Pull request by lorem--ipsum**
collections

### CODE REVIEW

File: src/client/components/global-event-listener/global-event-listener.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+  onMousedown(e: MouseEvent) {
+    if (this.props.mouseDown) this.props.mouseDown(e);
+  }
+
+  onKeydown(e: KeyboardEvent) {
+    if (this.props.escape && escapeKey(e)) this.props.escape();
+    if (this.props.enter && enterKey(e)) this.props.enter();
+    if (this.props.right && rightKey(e)) this.props.right();
+    if (this.props.left && leftKey(e)) this.props.left();
```
Comment (vogievetsky) [inline src/client/components/global-event-listener/global-event-listener.tsx]: I think you should feed the event through on all of these

File: src/client/components/modal/modal.tsx [hunk 2]
```diff
@@ -148,6 +139,11 @@ export class Modal extends React.Component<ModalProps, ModalState> {
 
     return <BodyPortal fullSize={true}>
       <div className={classNames('modal', className)}>
+        <GlobalEventListener
+          enter={this.onEnter.bind(this)}
+          escape={this.onEscape.bind(this)}
+          mouseDown={this.onMouseDown.bind(this)}
+        />
```
Comment (vogievetsky) [inline src/client/components/modal/modal.tsx]: very clean 👍

File: src/client/components/global-event-listener/global-event-listener.tsx (hunk 1 above)
Comment (lorem--ipsum) [inline src/client/components/global-event-listener/global-event-listener.tsx]: Good idea, will fix.

File: src/client/views/collection-view/collection-item-lightbox/collection-item-lightbox.scss [hunk 3]
```diff
@@ -35,16 +35,16 @@
...
     .modal-window {
       @extend %modal-cont;
-
-      width: 80%;
+      width: calc(100% - 270px);
 
       .headband {
-        padding: 10px 20px;
+        padding: 20px 20px;
+        //border-bottom: 1px solid $border-super-light;
```
Comment (lorem--ipsum) [inline src/client/views/collection-view/collection-item-lightbox/collection-item-lightbox.scss]: @asherbitter is this intentional ?

File: src/client/components/router/router.tsx [hunk 4]
```diff
@@ -20,9 +20,10 @@ import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import { $, Expression, Executor, Dataset } from 'plywood';
 import { Stage, Clicker, Essence, DataCube, Filter, Dimension, Measure } from '../../../common/models/index';
+import { replaceHash } from '../../utils/url/url';
 import { SvgIcon } from '../svg-icon/svg-icon';
 
-export interface RouteProps extends React.Props<any> { fragment: string; }
+export interface RouteProps extends React.Props<any> { fragment: string; alwaysShowOrphans?: boolean; }
```
Comment (vogievetsky) [inline src/client/components/router/router.tsx]: could you please make this multiline

File: src/client/components/router/router.tsx [hunk 5]
```diff
@@ -187,83 +219,137 @@ export class Router extends React.Component<RouterProps, RouterState> {
...
+      properties = this.extend(this.getPropertiesFromCrumbs(crumbs, fragment), properties);
+
       if (crumbs[0] === fragment || fragment.charAt(0) === ':') {
-        if (!(candidate.props.children instanceof Array)) {
-          return {fragment, route: candidate, crumbs};
-        } else if (crumbs.length === 1) {
-          return {fragment, route: candidate, crumbs};
+        let children = candidate.props.children;
+
+        if (!(children instanceof Array) || crumbs.length === 1) {
```
Comment (vogievetsky) [inline src/client/components/router/router.tsx]: Please use `Array.isArray` here

File: src/client/views/collection-view/collection-view.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        onAddItem={delegate.createItem.bind(this, collection)}
+      />
+
+     <div className="main-panel">
+       <Router onURLChange={this.onURLChange.bind(this)} rootFragment="collection">
+         <Route fragment=":collectionId" alwaysShowOrphans={true}>
+           <CollectionOverview collections={collections}/>
+
+           <Route fragment=":itemId">
+              <CollectionItemLightbox
```
Comment (vogievetsky) [inline src/client/views/collection-view/collection-view.tsx]: this seems to be indented by 3 spaces

**Comment by vogievetsky** [comment]
lol so many commits!

**Comment by vogievetsky** [comment]
gr 🎱

**Pull request merged**